13_import.sql

```diff
@@ -0,0 +1,575 @@
+    NULL::smallint AS outlet_2_material,
+    NULL::numeric(6, 3) AS outlet_2_clear_hight,
+    NULL::numeric(6, 3) AS outlet_2_depth_m,
+    (CASE WHEN EXISTS ( SELECT TRUE FROM qgep_import.manhole_quarantine q WHERE q.obj_id = ws.obj_id )
+    THEN TRUE
+    ELSE FALSE 
+    END) AS in_quarantine
+
+   FROM qgep_od.vw_qgep_wastewater_structure ws;
+
+
+-- create triggerfunction and trigger for mobile view
+
+CREATE OR REPLACE FUNCTION qgep_import.vw_manhole_insert_into_quarantine() RETURNS trigger AS $BODY$
+BEGIN
+  INSERT INTO qgep_import.manhole_quarantine
+  (
+  obj_id,
+  identifier,
+  situation_geometry,
+  co_shape,
+  co_diameter,
+  co_material,
+  co_positional_accuracy,
+  co_level,
+  _depth,
+  _channel_usage_current,
+  ma_material,
+  ma_dimension1,
+  ma_dimension2,
+  ws_type,
+  ma_function,
+  ss_function,
+  remark,
+  wn_bottom_level,
+  photo1,
+  photo2,
+  inlet_3_material,
+  inlet_3_clear_hight,
+  inlet_3_depth_m,
+  inlet_4_material,
+  inlet_4_clear_hight,
+  inlet_4_depth_m,
+  inlet_5_material,
+  inlet_5_clear_hight,
+  inlet_5_depth_m,
+  inlet_6_material,
+  inlet_6_clear_hight,
+  inlet_6_depth_m,
+  inlet_7_material,
+  inlet_7_clear_hight,
+  inlet_7_depth_m,
+  outlet_1_material,
+  outlet_1_clear_hight,
+  outlet_1_depth_m,
+  outlet_2_material,
+  outlet_2_clear_hight,
+  outlet_2_depth_m
+  )
+  VALUES
+  (
+  NEW.obj_id,
+  NEW.identifier,
+  NEW.situation_geometry,
+  NEW.co_shape,
+  NEW.co_diameter,
+  NEW.co_material,
+  NEW.co_positional_accuracy,
+  NEW.co_level,
+  NEW._depth,
+  NEW._channel_usage_current,
+  NEW.ma_material,
+  NEW.ma_dimension1,
+  NEW.ma_dimension2,
+  NEW.ws_type,
+  NEW.ma_function,
+  NEW.ss_function,
+  NEW.remark,
+  NEW.wn_bottom_level,
+  NEW.photo1,
+  NEW.photo2,
+  NEW.inlet_3_material,
+  NEW.inlet_3_clear_hight,
+  NEW.inlet_3_depth_m,
+  NEW.inlet_4_material,
+  NEW.inlet_4_clear_hight,
+  NEW.inlet_4_depth_m,
+  NEW.inlet_5_material,
+  NEW.inlet_5_clear_hight,
+  NEW.inlet_5_depth_m,
+  NEW.inlet_6_material,
+  NEW.inlet_6_clear_hight,
+  NEW.inlet_6_depth_m,
+  NEW.inlet_7_material,
+  NEW.inlet_7_clear_hight,
+  NEW.inlet_7_depth_m,
+  NEW.outlet_1_material,
+  NEW.outlet_1_clear_hight,
+  NEW.outlet_1_depth_m,
+  NEW.outlet_2_material,
+  NEW.outlet_2_clear_hight,
+  NEW.outlet_2_depth_m   
+  );
+  RETURN NEW;
+END; $BODY$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS on_mutation_make_insert ON qgep_import.vw_manhole;
+
+CREATE TRIGGER on_mutation_make_insert
+  INSTEAD OF INSERT OR UPDATE
+  ON qgep_import.vw_manhole
+  FOR EACH ROW
+  EXECUTE PROCEDURE qgep_import.vw_manhole_insert_into_quarantine();
+
+
+-- create triggerfunctions and triggers for quarantine table 
+
+CREATE OR REPLACE FUNCTION qgep_import.manhole_quarantine_try_structure_update() RETURNS trigger AS $BODY$
+DECLARE 
+  multi_situation_geometry geometry(MultiPoint,2056);
+BEGIN
+  multi_situation_geometry = st_collect(NEW.situation_geometry)::geometry(MultiPoint,2056);
```

Comment:
idem

Reply:
This does not work:
```
LINE 3:   multi_situation_geometry geometry(MultiPoint,:SRID);
                                   ^
CONTEXT:  invalid type name "geometry(MultiPoint,:SRID)"
```
I don't understand why because the one on line 9 works perfectly through.

Reply:
that's because it's a function and not some simple code.
you'll have to do something like
```
EXECUTE format($TRIGGER$
CREATE OR REPLACE FUNCTION qgep_import.manhole_quarantine_try_structure_update() RETURNS trigger AS $BODY$
+DECLARE 
+  multi_situation_geometry geometry(MultiPoint,2056);
+BEGIN
+  multi_situation_geometry = st_collect(NEW.situation_geometry)::geometry(MultiPoint, %s);
......
......
$TRIGGER$, :SRID);
```

might need a few tries ;)

Reply:
Hm... It seems that `EXECUTE format(` is only working in a `DO`-block and in a `DO`-block the variables aren't available as well. I'll continue trying tomorrow and otherwise put everything into python...

Reply:
well, nothing prevents to add a `DO` block in the SQL file (that was not possible in older versions)

Reply:
I struggled a bit but did not find a solution.
Let's see if something comes out https://stackoverflow.com/questions/51192840/use-a-variable-in-a-postgresql-execute-statement

Reply:
Let's go with 2056 for now if there's no easy solution

Reply:
Following the reply on SO, here is what works:
```PLpgSQL
SELECT set_config('my.srid', :SRID::text, false);
DO $DO$
BEGIN
EXECUTE format($TRIGGER$
  CREATE OR REPLACE FUNCTION qgep_od.my_trigger_fcn() RETURNS trigger AS
  $BODY$
    DECLARE
      multi_situation_geometry geometry(MultiPoint,%1$s);
    BEGIN
      multi_situation_geometry = st_collect(NEW.situation_geometry)::geometry(MultiPoint,%1$s);
      .........
      .........
      .........
    END;
  $BODY$
  LANGUAGE plpgsql;
$TRIGGER$, current_setting('my.srid'));
END
$DO$;
```

Reply:
Used that.